chunk/schema.go

```diff
@@ -343,46 +118,101 @@ func v6Schema(cfg SchemaConfig) Schema {
 	}
 }
 
+// v7 schema is an extension of v6, with support for queries with no metric names
+func v7Schema(cfg SchemaConfig) Schema {
+	return schema{
+		cfg.dailyBuckets,
+		v7Entries{},
+	}
+}
+
 // schema implements Schema given a bucketing function and and set of range key callbacks
 type schema struct {
-	buckets func(from, through model.Time, userID string, metricName model.LabelValue, callback bucketCallback) ([]IndexEntry, error)
+	buckets func(from, through model.Time, userID string) []Bucket
 	entries entries
 }
 
 func (s schema) GetWriteEntries(from, through model.Time, userID string, metricName model.LabelValue, labels model.Metric, chunkID string) ([]IndexEntry, error) {
-	return s.buckets(from, through, userID, metricName, func(bucketFrom, bucketThrough uint32, tableName, hashKey string) ([]IndexEntry, error) {
-		return s.entries.GetWriteEntries(bucketFrom, bucketThrough, tableName, hashKey, labels, chunkID)
-	})
+	var result []IndexEntry
+
+	buckets := s.buckets(from, through, userID)
+	for _, bucket := range buckets {
```

Comment:
Nit, but IMO this could be `for _, bucket := range s.buckets(from, through, userID) {`